localstack-core/localstack/services/apigateway/next_gen/execute_api/template_mapping.py

```diff
@@ -0,0 +1,218 @@
+            if len(obj.strip()) == 0:
+                return "{}"
+            return json.dumps(obj)[1:-1]
+        if obj in (True, False):
+            return str(obj).lower()
+        return str(obj)
+
+    def parseJson(self, s: str):
+        obj = json.loads(s)
+        return AttributeDict(obj) if isinstance(obj, dict) else obj
+
+
+class VelocityInput:
+    """
+    Simple class to mimic the behavior of variable '$input' in AWS API Gateway integration
+    velocity templates.
+    See: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
+    """
+
+    def __init__(self, body, params):
+        self.parameters = params or {}
+        self.value = body
+
+    def path(self, path):
+        if not self.value:
+            return {}
+        value = self.value if isinstance(self.value, dict) else json.loads(self.value)
+        return extract_jsonpath(value, path)
+
+    def json(self, path):
+        path = path or "$"
+        matching = self.path(path)
+        if isinstance(matching, (list, dict)):
+            matching = json_safe(matching)
+        return json.dumps(matching)
+
+    @property
+    def body(self):
+        return self.value
+
+    def params(self, name=None):
+        if not name:
+            return self.parameters
+        for k in ["path", "querystring", "header"]:
+            if val := self.parameters.get(k).get(name):
+                return val
+        return ""
+
+    def __getattr__(self, name):
+        return self.value.get(name)
+
+    def __repr__(self):
+        return "$input"
+
+
+class ApiGatewayVtlTemplate(VtlTemplate):
+    """Util class for rendering VTL templates with API Gateway specific extensions"""
+
+    def prepare_namespace(self, variables) -> dict[str, Any]:
+        namespace = super().prepare_namespace(variables)
+        input_var = variables.get("input") or {}
+        variables = {
+            "input": VelocityInput(input_var.get("body"), input_var.get("params")),
+            "util": VelocityUtilApiGateway(),
+        }
+        namespace.update(variables)
+        return namespace
+
+    def render_request(
+        self, template: str, variables: MappingTemplateVariables
+    ) -> tuple[str, ContextVarsRequestOverride]:
+        vars: MappingTemplateVariables = copy.deepcopy(variables)
+        vars["context"]["requestOverride"] = ContextVarsRequestOverride(
+            querystring={}, header={}, path={}
+        )
+        result = self.render_vtl(template=template.strip(), variables=vars)
+        return result, vars["context"]["requestOverride"]
+
+    def render_response(
+        self, template: str, variables: MappingTemplateVariables
+    ) -> tuple[str, ContextVarsResponseOverride]:
+        pass
+
+    # TODO Maybe we don't need those methods and they should belong on the integration response handler?
+    #  And we should raise the appropriate exception from the gateway responses.
```

Comment:
Agreed here, I think we will update this, I wonder when it is actually executed. Interesting find!

Reply:
They were part on the ResponseTemplate. It was rendering and then trying to load the Json. But I feel like it shouldn't be a responsibility of the template, but instead of the integration response.

Reply:
Do you think there's a similar thing for IntegrationRequest, and that it checks if the rendered body is valid or not before sending it down?

Reply:
I will have to double check with AWS, but our current implementation was raising an error

Reply:
I double checked for the requestTemplate only.
Everything is sent as is. except for GET, OPTIONS, HEAD, an empty string is sent instead.

I updated the handler to reflect